### ISSUE TITLE
[BUG] Swap failed to execute

### ISSUE THREAD

**Issue by vldkhh** [GoodDollar/GoodCollective#96]
https://celoscan.io/tx/0x9e11d82ee3146448b0f08ab2cba6c89ab850e8fb3ca19908a6ec0b01b0044d2d

![image](https://github.com/GoodDollar/GoodCollective/assets/6606028/fb4ef74f-a42a-4ad5-a7b0-1dc5b7ce8131)

**Comment by krisbitney** [comment]
Could you tell me what the revert reason was? The Celo explorer doesn't tell me what the error is and i cannot view the images you shared on GitHub.

**Comment by krisbitney** [comment]
@sirpy I have noticed that sometimes transactions fail for a couple of reasons:
1.  `estimateGas` sometimes fails
2. unknown on-chain exceptions

These failures can occur with any of the support functions, not just swaps.

For example, I attempted to use `supportFlow` with the following parameters I got an uninformative on-chain exception:
````
flowRate:  115740740740740 // 10 G$ for duration of 1 day
collective:  0xe31eb27dc0ec9207b7f114faff46bbb5c962ae0b
```
<img width="558" alt="Screenshot 2024-01-13 at 8 24 14 PM" src="https://github.com/GoodDollar/GoodCollective/assets/39842820/c5b394a1-40c8-4ecc-b8d4-434aa9592ce0">
